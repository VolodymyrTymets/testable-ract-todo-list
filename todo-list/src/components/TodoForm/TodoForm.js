import React from 'react';
import PropTypes from 'prop-types';
import styles  from './TodoForm.module.scss'

const ToDoForm = ({ test }) => {
  return (
    <form className={styles['todo-form']} data-test="form">
      <input type="text" data-test="input" />
      <buttom type="submit" data-test="button" >Submit</buttom>
    </form>
  );
};

ToDoForm.propTypes = {
  test: PropTypes.string.isRequired,
};

export default ToDoForm;
