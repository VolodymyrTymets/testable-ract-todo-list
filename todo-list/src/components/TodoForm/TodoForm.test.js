import React from 'react';
import checkPropTypes from 'check-prop-types';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TodoForm from './TodoForm';

const requiredProps = {
  test: 'test',
};

const setup = (props) => shallow(
  <TodoForm {...props} {...requiredProps} />);

describe("<TodoForm /> - render", () => {
  it("should render without error", () => {
    const wrapper = setup();
    // if has not snapshots
    expect(wrapper.length).not.toBe(0);
    // if has snapshots
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should not throw warning with expected props", () => {
    const props = {
      test: 'test'
    };
    // todo move to utils
    const propError = checkPropTypes(
      TodoForm.propTypes,
      props,
      'prop',
      TodoForm.name);
    expect(propError).toBeUndefined();
  })
});


// todo: move to utils
const findById = (component, id) => component.find(`[data-test="${id}"]`);

describe("<TodoForm /> - components", () => {
  it("should contains form, input, button", () => {
    const wrapper = setup();

    const form = findById(wrapper, 'form');
    const input = findById(wrapper, 'input');
    const button = findById(wrapper, 'button');

    expect(form.length).not.toBe(0);
    expect(input.length).not.toBe(0);
    expect(button.length).not.toBe(0);
  })
});

// todo check snapshot sate for each test
// todo check Mock functions here https://medium.com/codeclan/testing-react-with-jest-and-enzyme-20505fec4675