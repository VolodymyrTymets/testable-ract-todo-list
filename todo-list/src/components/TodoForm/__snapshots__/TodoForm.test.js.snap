// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> - render should render without error 1`] = `
<form
  className="todo-form"
  data-test="form"
>
  <input
    data-test="input"
    type="text"
  />
  <buttom
    data-test="button"
    type="submit"
  >
    Submit
  </buttom>
</form>
`;
